### DISCUSSION TITLE
Autoloads should be a 5th main top tab before AssetLib

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#10019] [Editor]
REQUEST:  Button to open autoloads page in project settings.

WHERE:  On main script window, top right, near the open docs stuff.

OR 

Autolaods are so important right now maybe they should be a 5th main tab before AssetLib.

WHY: People are using autoloads for global everything, audio, saving/loading etc. 50%+ of code can be in autoloads and you work on them in...project settings?!? Bring them out.

**Comment by theraot** [comment]
Yes they are in project settings. I understand that people might use many autoloads, but are they really adding them and removing them all the time during development?

If yes, perhaps we should talk about making adding an option to the context menu of the FileSystem to add/remove from autoloads directly.

---

It seems to me that it is more common to want to edit existing autoloads than to add and remove. So I recognize and understand the desire to have quick access to autoloads for those who use them a lot. However, I'm not convinced another main tab is a good idea.

Consider that you might organize your project so you have your autolaods in a distinct folder in the FileSystem, and you might put that folder in favotires.

*In fact, now that you bring the topic, perhaps FileSystem could show autoloads like it shows Favorites.*

---

By the way, see also:

- https://github.com/godotengine/godot-proposals/issues/4993

**Reply by shak2** [reply]
A small button in the top right of the scripts panel will solve this better then. Near the open docs link in the top right. The button should open the autoloads page in project settings.

Basically need a quicker way to open autoloads scripts for editing when needed.

**Reply by theraot** [reply]
Hmm... Given that scenes could also be autoloads, at least that explains why you might want it on top next to AssetLib.
